### PR TITLE
E2E: Add commands to update system config and clear whitelisted cookies

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Added the following commands:
- `apiUpdateConfig` to update system config.
  - Unified other command/test to use this
  - Added [merge-deep](https://www.npmjs.com/package/merge-deep) as dev dependencies to easily merge config
- ~~`clearWhiteListedCookies` to clear whitelisted cookies - this fixes issue on test cases where it expects to have a user fully logout~~ as discussed with Bob, clearing whitelisted cookies are done via apiLogout

#### Ticket Link
none

### CODE REVIEW

File: cypress/integration/onboarding/email/existing_email_adress_spec.js [hunk 1]
```diff
@@ -33,16 +33,12 @@ function signupWithEmail(name, pw) {
 
 describe('Email Address', () => {
     before(() => {
-        // Before test, login as sysadmin, set Enable Open Server to true then logout
-        cy.apiLogin('sysadmin');
-        cy.apiEnableOpenServer(true);
-        cy.apiLogout();
-    });
-
-    after(() => {
```
Comment (saturninoabril) [inline cypress/integration/onboarding/email/existing_email_adress_spec.js]: after hook here is not necessary

**Comment by thekiiingbob** [comment]
@saturninoabril You may want to double check depending on what issue this fixes, but https://github.com/mattermost/mattermost-webapp/pull/2836 may address the whitelisted cookies issue as they would no longe be whitelisted and should not persist the same way.

File: cypress/support/api_commands.js [hunk 2]
```diff
@@ -317,3 +303,28 @@ Cypress.Commands.add('apiUnpinPosts', (postId) => {
         method: 'POST',
     });
 });
+
+// *****************************************************************************
+// System config
+// *****************************************************************************
+
+Cypress.Commands.add('apiUpdateConfig', (newSettings = {}) => {
```
Comment (thekiiingbob) [inline cypress/support/api_commands.js]: 👍

File: cypress/support/ui_commands.js [hunk 3]
```diff
@@ -388,3 +388,13 @@ Cypress.Commands.add('updateChannelHeader', (text) => {
         type('{enter}').
         wait(TIMEOUTS.TINY);
 });
+
+// ************************************************************
+// Cookie
+// ************************************************************
+
+Cypress.Commands.add('clearWhiteListedCookies', () => {
```
Comment (thekiiingbob) [inline cypress/support/ui_commands.js]: Should this just be a part of `apiLogout`? I think we'd want to do it everytime we logout, but maybe not.

File: cypress/integration/onboarding/email/existing_email_adress_spec.js [hunk 4]
```diff
@@ -60,6 +56,7 @@ describe('Email Address', () => {
 
         // 8. Logout and signup another user with the same email but different username and password
         cy.apiLogout();
+        cy.clearWhiteListedCookies();
```
Comment (thekiiingbob) [inline cypress/integration/onboarding/email/existing_email_adress_spec.js]: Per my comment below, I think we should clear the cookies as part of api logout by default.

File: cypress/support/ui_commands.js (hunk 3 above)
Comment (saturninoabril) [reply cypress/support/ui_commands.js]: Sure, that makes sense, I'll update.

**Comment by saturninoabril** [comment]
@thekiiingbob Yeah, I've seen and tried [that](https://github.com/mattermost/mattermost-webapp/pull/2836/files#diff-2b07e229b86aa9d270302ce5fcdf29a1R12) but it didn't solve the issue.

**Comment by saturninoabril** [comment]
@thekiiingbob PR updated

**Pull request merged**